src/xcode/ENA/ENA/Source/Scenes/TraceLocations/Details/TraceLocationDetailsViewController.swift

```diff
@@ -58,25 +58,33 @@ class TraceLocationDetailsViewController: UIViewController, FooterViewHandling {
 			onDuplicateButtonTap(viewModel.traceLocation)
 		}
 	}
-	
+
+	// MARK: - Private
+
+	private let viewModel: TraceLocationDetailsViewModel
+
+	private let onPrintVersionButtonTap: (PDFView) -> Void
+	private let onDuplicateButtonTap: (TraceLocation) -> Void
+	private let onDismiss: () -> Void
+	private var subscriptions = [AnyCancellable]()
+
 	private func generateAndPassQRCodePoster() {
 		viewModel.fetchQRCodePosterTemplateData { [weak self] templateData in
 			switch templateData {
 			case let .success(templateData):
 				DispatchQueue.main.async { [weak self] in
 					self?.footerView?.setLoadingIndicator(false, disable: false, button: .primary)
-
 					do {
 						let pdfView = try self?.createPdfView(templateData: templateData)
 						self?.onPrintVersionButtonTap(pdfView ?? PDFView())
 					} catch {
+						self?.footerView?.setLoadingIndicator(false, disable: false, button: .primary)
```

Comment:
I don't see why it is necessary to set the here again. It's the same as in line 67 76